src/config.rs

```diff
@@ -15,12 +15,63 @@ use trussed::{
     Client,
 };
 
+#[derive(Debug)]
+/// Structure meant to be stored in  a `static` to signal applications that they have been factory-resetted by the admin app
+pub struct ResetSignalAllocation(AtomicU8);
+
+impl ResetSignalAllocation {
+    pub fn load(&self) -> ResetSignal {
+        let v = self.0.load(Ordering::Relaxed);
+        ResetSignal::from_repr(v).expect("A reset signal value")
+    }
+
+    pub fn set_factory_reset(&self) -> bool {
+        self.0
+            .compare_exchange(
+                ResetSignal::None as u8,
+                ResetSignal::FactoryReset as u8,
+                Ordering::Relaxed,
+                Ordering::Relaxed,
+            )
+            .is_ok()
+    }
+
+    pub fn set_config_changed(&self) {
+        self.0
+            .store(ResetSignal::ConfigChanged as u8, Ordering::Relaxed)
+    }
+
+    /// Factory reset can be acknowledged so that the application can restart working
+    ///
+    /// A configuration change cannot be acknowledged as it requires a power cycle to be taken into account.
+    pub fn ack_factory_reset(&self) {
+        self.0
+            .store(ResetSignal::ConfigChanged as u8, Ordering::Relaxed)
+    }
```

Comment:
I’m not sure I understand this one.  Why are we in the `ConfigChanged` state after the factory reset?  Shouldn’t we check whether we are actually in the `FactoryReset` state?